### ISSUE TITLE
🐛 BUG: `wrangler pages dev` is returning an old D1Result object shape

### ISSUE THREAD

**Issue by tom-sherman** [cloudflare/workers-sdk#2673]
### Which Cloudflare product(s) does this pertain to?

D1, Pages

### What version of `Wrangler` are you using?

2.9.0

### What operating system are you using?

Mac

### Describe the Bug

When running `wrangler pages dev` (with local flag on or off) results from D1 are returned in the old shape ie.

```ts
declare type D1Result<T = unknown> = {
  results?: T[];
  lastRowId: number | null;
  changes: number;
  duration: number;
  error?: string;
};
```

However when deployed the function picks up the new [D1 result shape](https://github.com/cloudflare/workerd/blob/42d45ba4cd41ed714b737a605a1022cc8d24cdde/types/defines/d1.d.ts#L1-L6):

```ts
type D1Result<T> = {
  success: boolean;
  data: T[];
  meta: {
    duration: number;
    last_row_id: number | null;
    changes: null;
    served_by: string;
    internal_stats: null;
  };
}
```

This is causing my pages to work fine locally, but fail when deployed (or vice versa, if I correct the code to use the new shape).

I don't have a minimal repro, but this is happening on my blog which is open source: https://github.com/tom-sherman/tom-sherman.com/pull/26/files

**Comment by Skye-31** [comment]
This should be fixed on wrangler 2.10.0 🙂

**Comment by tom-sherman** [comment]
Yup, it works thanks!
